modules/cluster-cloud-controller-config-osp.adoc

```diff
@@ -0,0 +1,271 @@
+= The {rh-openstack} Cloud Controller Manager (CCM) config file
+
+A {rh-openstack-first} CCM config file, called `cloud.conf` by convention, comprises a number of options in an INI format.
+
+Except when indicated otherwise, these options are mandatory for clusters that run on {rh-openstack}.
+
+[id="ccm-config-global-options"]
+== Global options
+
+The following options are used for {rh-openstack} CCM authentication with the {rh-openstack} Identity service, also known as Keystone. They are similiar to the global options that you can set by using the `openstack` CLI.
+
+|===
+| Option | Description
+
+| `auth-url`
+| The {rh-openstack} Identity service URL. For example, `http://128.110.154.166/identity`.
+
+| `os-endpoint-type`
+| Optional. The type of endpoint to use from the service catalog. If unset, public endpoints are used.
+
+| `ca-file`
+| Optional. The CA certificate bundle file for communication with the {rh-openstack} Identity service. If you use the HTTPS protocol with The Identity service URL, this option is required.
+
+| `cert-file`
+| Optional. The client certificate path to use for client TLS authentication.
+
+| `key-file`
+| Optional. The client private key path to use for client TLS authentication.
+
+| `username`
+| The Identity service user name. Leave this option unset if you are using Identity service application credentials.
+
+| `password`
+| The Identity service user password. Leave this option unset if you are using Identity service application credentials.
+
+| `region`
+| The Identity service region name.
+
+| `domain-id`
+| The Identity service user domain ID. Leave this option unset if you are using Identity service application credentials.
+
+| `domain-name`
+| The Identity service user domain name. This option is not required if you set `domain-id`.
+
+| `tenant-id`
+| The Identity service project ID. Leave this option unset if you are using Identity service application credentials.
+
+If you are using version 3 of the Identity API, which changed the identifier `tenant` to `project`, the value of `tenant-id` is automatically mapped to the project construct in the API. 
+
+| `tenant-name`
+| The Identity service project name. You can ignore this option if `tenant-id` is set.
+
+| `tenant-domain-id`
+| The Identity service project domain ID.
+
+| `tenant-domain-name`
+| The Identity service project domain name.
+
+| `user-domain-id`
+| The Identity service user domain ID.
+
+| `user-domain-name`
+| The Identity service user domain name.
+
+| `trust-id`
+| The Identity service trust ID. A trust represents the authorization of a user, or trustor, to delegate roles to another user, or trustee. Optionally, a trust authorizes the trustee to impersonate the trustor. You can find available trusts by querying the `/v3/OS-TRUST/trusts` endpoint of the Identity service API.
+
+| `trustee-id`
+| The Identity service trustee user ID.
+
+| `trustee-password`
+| The Identity service trustee user password.
+
+| `use-clouds`
```

Comment:
@MaysaMacedo We are always testing with below content:

    [Global]
    region      = regionOne
    ca-file     = /etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem
    use-clouds  = true
    clouds-file = /etc/openstack/secret/clouds.yaml
    cloud       = openstack

If I understood correctly, this configMap is not expected to be changed by the user, so... in which case we can have different configuration than above?

Reply:
@MaysaMacedo Any thoughts on this?

Reply:
Since we're enforcing the UseClouds to true in the code, I believe there is no need to document the following options: auth-url, os-endpoint-type, username, password, domain related ones, tenant ones and user ones.

@stephenfin @dulek what do you think?

Reply:
@stephenfin @dulek ^^^

Reply:
I agree with Maysa here.